subsys/mgmt/hawkbit/hawkbit_device.h

```diff
@@ -18,6 +18,8 @@
 #define DEVICE_ID_HEX_MAX_SIZE	((DEVICE_ID_BIN_MAX_SIZE * 2) + 1)
 #endif
 
+#define CONTROLLER_ID_MAX_SIZE	(32 + DEVICE_ID_HEX_MAX_SIZE)
```

Comment:
include it instead in the DEVICE_ID_HEX_MAX_SIZE for the non custom dev id and use sizeof(CONFIG_BOARD) instead of the arbitrary 32 and don't have a seperate CONTROLLER_ID_MAX_SIZE